debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/events/BufferingChangeStreamCursor.java

```diff
@@ -198,7 +224,8 @@ public void run() {
                 noMessageIterations = 0;
                 fetchEvents(cursor);
             }
-            catch (InterruptedException e) {
+            catch (Throwable e) {
+                error.set(e);
                 throw new DebeziumException("Fetcher thread interrupted", e);
             }
```

Comment:
This needs to stay as two separate branches. With `InterruptedException` we want to end right away. Whereas on other exceptions we want to keep the opportunity to drain the current buffer queue. 

So we should set the error and mark the fetcher as not running.